src/layouts/slab.py

```diff
@@ -66,44 +69,125 @@ class NodeType(IntEnum):
 ORDER_BOOK_LAYOUT = cStruct(Padding(5), "account_flags" / ACCOUNT_FLAGS_LAYOUT, "slab_layout" / SLAB_LAYOUT, Padding(7))
 
 
+class SlabHeader(NamedTuple):
+    bump_index: int
+    free_list_length: int
+    free_list_root: int
+    root: int
+    leaf_count: int
+
+
+@dataclass
+class SlabNode:
+    pass
+
+
+@dataclass
+class SlabUninitializedNode(SlabNode):
+    pass
+
+
+@dataclass
+class SlabLeafNode(SlabNode):
+    owner_slot: int
+    fee_tier: int
+    key: int
+    owner: PublicKey
+    quantity: int
+    client_order_id: int
+
+
+@dataclass
+class SlabInnerNode(SlabNode):
+    prefix_len: int
+    key: int
+    children: List[int]
+
+
+@dataclass
+class SlabFreeNode(SlabNode):
+    next: int
+
+
+@dataclass
+class SlabLastFreeNode(SlabNode):
+    pass
+
+
+def convert_construct_node_to_class(construct_nodes) -> List[SlabNode]:
+    res: List[SlabNode] = []
+    for construct_node in construct_nodes:
+        node_type = construct_node.tag
+        node = construct_node.node
+        if node_type == NodeType.UNINTIALIZED:
+            res.append(SlabUninitializedNode())
+        elif node_type == NodeType.LEAF_NODE:
+            res.append(
+                SlabLeafNode(
+                    owner_slot=node.owner_slot,
+                    fee_tier=node.fee_tier,
+                    key=int.from_bytes(node.key, "little"),
+                    owner=PublicKey(node.owner),
+                    quantity=node.quantity,
+                    client_order_id=node.client_order_id,
+                )
+            )
+        elif node_type == NodeType.INNER_NODE:
+            res.append(
+                SlabInnerNode(
+                    prefix_len=node.prefix_len, key=int.from_bytes(node.key, "little"), children=node.children
+                )
+            )
+        elif node_type == NodeType.FREE_NODE:
+            res.append(SlabFreeNode(next=node.next))
```

Comment:
Should this be `next` be set to `node` or `node.next`?

Reply:
`next` is the next index in the free list that contains the free node, so it should be `node.next. I think.

Reply:
Oh. I missed that `next` is an `int`.